code/workspaces/infrastructure-api/resources/jupyter.configmap.template.yml

```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ configMapName }}
+data:
+  config: |
+    # From https://github.com/jupyterhub/jupyter-server-proxy/blob/master/jupyter_server_proxy/config.py
+    # Access to Dask schedulers is also controlled with network policies.
+    # This allows the Dask cluster to be seen from the Dask JupyterLab extension
+    # with address proxy/dask-scheduler-<cluster name>:8787
+    def host_allowlist(handler, host):
+      return host in ["localhost", "127.0.0.1"] or host.startswith("dask-scheduler-")
```

Comment:
Should we be making the hostname fully qualified just to be sure? Don't want to over complicate it if that makes it much trickier.

Reply:
Actually, the opposite is true - we don't want need a fully qualified hostname, because the cluster must be in the same namespace as the notebook - and a network policy prevents you from accessing clusters outside the project namespace.  So fully qualifying is allowed, but redundant.

Reply:
True, but someone could potentially proxy through to something like `dask-scheduler-test.hacker.com` using this method, which is presumably why the security on this function is strictly controlled.

Reply:
I've modified this so that if it starts with 'dask-scheduler-', it can't contain any "." characters, and so only proxies within the same namespace can now be made (and this is also enforced by the network policy).